packages/gestalt/src/TableRowExpandable.js

```diff
@@ -0,0 +1,51 @@
+// @flow strict
+import React, { type Node, useState } from 'react';
+import cx from 'classnames';
+import styles from './Table.css';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+
+type Props = {|
+  children: Node,
+  hoverStyle?: 'gray' | 'none',
+  expandedContents: Node,
+|};
+
+export default function TableRowExpandable(props: Props): Node {
+  const [expanded, setExpanded] = useState(false);
+  const [hovered, setHover] = useState(false);
+  const handleHover: (isHovered: boolean) => void = (isHovered: boolean) =>
+    setHover(isHovered);
+  const hoverStyle = props.hoverStyle || 'gray';
+  const cs =
+    hovered && hoverStyle === 'gray'
+      ? cx(styles.hoverShadeGray)
+      : cx(styles.hoverNone);
+
+  return (
+    <>
+      <tr
+        className={cs}
+        onMouseEnter={() => handleHover(true)}
+        onMouseLeave={() => handleHover(false)}
+      >
+        <td>
+          <IconButton
+            accessibilityLabel=""
+            icon={expanded ? 'arrow-up' : 'arrow-down'}
+            iconColor="darkGray"
+            onClick={() => setExpanded(!expanded)}
+          />
+        </td>
+        {props.children}
+      </tr>
+      {expanded && (
+        <tr>
+          <td colSpan={Object.keys(props.children).length}>
```

Comment:
@faisalgedi Are we sure this logic is correct? In some of the examples, the colspan should be 1 higher

![image](https://user-images.githubusercontent.com/127199/95496313-7c775e80-0955-11eb-81c8-627b77ff0db2.png)